packages/riverpod/lib/src/core/ref.dart

```diff
@@ -139,8 +139,19 @@ final <yourProvider> = Provider(dependencies: [<dependency>]);
   /// If [keepAlive] is invoked multiple times, all [KeepAliveLink] will have
   /// to be closed for the provider to dispose itself when all listeners are removed.
   ///
-  /// Finally, keep in mind that [keepAlive] doesn't mark a provider immune to rebuilds or invalidations.
-  /// Indeed, when a provider disposes, **all** its associated [KeepAliveLink]s will be closed.
+  /// Keep in mind that [keepAlive] doesn't mark a provider immune to rebuilds or invalidations.
+  /// Indeed, [keepAlive] has no effect on disposal through [watch] or [invalidate].
+  ///
+  /// Furthermore, when a provider disposes, all its associated [KeepAliveLink]s will be closed.
+  ///
+  /// The most extreme scenario is a provider with no active listeners,
+  /// which rebuilds for whatever reason, causing it to dispose
+  /// and to close any active [KeepAliveLink]:
+  /// in such scenario, the provider won't recompute at all.
+  ///
+  /// In other words, without listeners, it might be that
```

Comment:
I think this part is a bit redundant and can be removed.
@rrousselGit What do you think?